src/ui/dialogs/contactsearch.ts

```diff
@@ -0,0 +1,278 @@
+
+        html+='<td>';
+        if ($(items[i]).children('first').length>0)
+        {
+            html+=$(items[i]).children('first').text();
+        }
+        html+='</td><td>';
+        if ($(items[i]).children('last').length>0)
+        {
+            html+=$(items[i]).children('last').text();
+        }
+        html+='</td><td>';
+        if ($(items[i]).children('nick').length>0)
+        {
+            html+=$(items[i]).children('nick').text();
+        }
+        html+='</td><td>';
+        if ($(items[i]).children('email').length>0)
+        {
+            html+=$(items[i]).children('email').text();
+        }
+        html+='</td>';
+
+        html+='</tr>';
+    }
+
+    html+='</tbody></table>';
+
+    return html;
+}
+
+function getTableForExtendedResults(header, items, fields)
+{
+    let html = '<table id="jsxc-searchresults" class="jsxc-searchresults-table"><thead class="jsxc-searchresults-table-head"><tr class="jsxc-searchresults-table-head-row">';
+
+    for (let headitm of header)
+    {
+        html+='<th>'+headitm+'</th>';
+    }
+    html+='</tr></thead><tbody class="jsxc-searchresults-table-body">';
+
+    for (let i=0;i<items.length;i++)
+    {
+        html+='<tr class="jsxc-searchresults-table-body-row '+(i%2===0?'jsxc-searchresults-odd':'jsxc-searchresults-even')+'"><td><input class="jsxc-searchresults-add" type="checkbox" name="jsxc-add-contact" value="'+i+'"/></td>';
+
+            for (let field of fields)
+            {
+                html+='<td>';
+                html+=$(items[i]).find('field[var="'+$(field).attr('var')+'"]').text();
+                html+='</td>';
+            }
+
+        html+='</tr>';
+    }
+
+    html+='</tbody></table>';
+
+    return html;
+}
+
+function showForm(form: Form, connection: IConnection, jid : JID) {
+   let formElement = form.toHTML();
+
+   let submitButton = $(`<div class="form-group">
+      <div class="col-sm-offset-4 col-sm-8">
+         <button class="jsxc-button jsxc-button--default jsxc-js-close" type="button">${Translation.t('Cancel')}</button>
+         <button class="jsxc-button jsxc-button--primary" type="submit">${Translation.t('Search')}</button>
+      </div>
+   </div>`);
+   formElement.append(submitButton);
+
+   dialog.getDom().find('#jsxc-search-content').empty().append(formElement);
+
+   return new Promise((resolve, reject) => {
+        formElement.submit((ev) => {
+        ev.preventDefault();
+
+        let formval = Form.fromHTML(formElement.get(0));
+
+        let submitPromise = connection.getSearchService().executeSearchForm(jid,formval).then((stanza) => {
+            return stanza;
+        });
+        resolve(submitPromise);
+      });
+
+      formElement.find('.jsxc-js-close').click((ev) => {
+         ev.preventDefault();
+
+         dialog.close();
+
+         resolve(null);
+      });
```

Comment:
The dialog close is looking for `.jsxc-js-close` and adding the close handler. So this can be removed.

Reply:
without it does not close the dialog... maybe because its an dynamic form..